### PR TITLE
New package: faust-2.20.2

### PR DESCRIPTION
**Pull request by DavideMaggio**

### CODE REVIEW

File: srcpkgs/faust/template [hunk 1]
```diff
@@ -0,0 +1,51 @@
+# Template file for 'faust'
+pkgname=faust
+version=2.20.2
+revision=1
+build_style=gnu-makefile
+hostmakedepends="cmake pkg-config libmicrohttpd-devel llvm llvm-libunwind-devel"
+short_desc="Functional Programming Language for Real Time Signal Processing"
+maintainer="mag <mag-one@autistici.org>"
+license="GPL-2.0-or-later"
+homepage="http://www.faust.grame.fr"
```
Comment (Piraty) [inline srcpkgs/faust/template]: in fact it is `http://faust.grame.fr/`

File: srcpkgs/faust/template [hunk 2]
```diff
@@ -0,0 +1,51 @@
+# Template file for 'faust'
+pkgname=faust
+version=2.20.2
+revision=1
+build_style=gnu-makefile
+hostmakedepends="cmake pkg-config libmicrohttpd-devel llvm llvm-libunwind-devel"
```
Comment (Piraty) [inline srcpkgs/faust/template]: `*-devel` dependencies go into `makedepends=`, as they're probably target specific

File: srcpkgs/faust/template [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+short_desc="Functional Programming Language for Real Time Signal Processing"
+maintainer="mag <mag-one@autistici.org>"
+license="GPL-2.0-or-later"
+homepage="http://www.faust.grame.fr"
+distfiles=https://github.com/grame-cncm/faust/releases/download/${version}/${pkgname}-${version}.tar.gz
+checksum="c32588ee60d0f7ca25034d3d5436def6b35495f4d3beb3b02208d48684bcfbe9"
+nostrip_files="libsndfile.so "
+noshlibprovides="libsndfile.so"
+
+case $XBPS_MACHINE in
```
Comment (Piraty) [inline srcpkgs/faust/template]: you may want to test for `$XBPS_TARGET_MACHINE` instead, and add libexecinfo-devel to  `makedepends=` (not host...)

**Comment by Piraty** [comment]
thanks for your first contribution.
please mark the PR as [WIP] or "draft" while you work on it and squash your commits

**Comment by ndowens** [comment]
Here is a cleaner way to do this: 
https://github.com/void-linux/void-packages/compare/master...ndowens:faust?expand=1

**Comment by DavideMaggio** [comment]
> thanks for your first contribution.
> please mark the PR as [WIP] or "draft" while you work on it and squash your commits

Thanks a lot for all your suggestions.
Now I'll try to squash my commits....

File: srcpkgs/faust/template [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+# rm -rf $PKGDESTDIR/usr/share/faust/android/
+#}
+
+# BETTER (?) WORKAROUND:
+#
+skiprdeps="/usr/lib/faust/android/app/lib/libsndfile/lib/armeabi-v7a/libsndfile.so /usr/lib/faust/android/app/lib/libsndfile/lib/arm64-v8a/libsndfile.so"
+
+post_install() {
+	mv $PKGDESTDIR/usr/share/faust $PKGDESTDIR/usr/lib
+	ln -s /usr/lib/faust $PKGDESTDIR/usr/share/faust
```
Comment (Piraty) [inline srcpkgs/faust/template]: I'd say move them to /usr/lib instead of symlinking

File: srcpkgs/faust/template [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+homepage="http://www.faust.grame.fr/"
+distfiles=https://github.com/grame-cncm/faust/releases/download/${version}/${pkgname}-${version}.tar.gz
+checksum="c32588ee60d0f7ca25034d3d5436def6b35495f4d3beb3b02208d48684bcfbe9"
+nostrip_files="libsndfile.so "
+noshlibprovides="libsndfile.so"
+
+case $XBPS_TARGET_MACHINE in
+	*-musl) makedepends+=" libexecinfo-devel";;
+	#armv6l-musl) broken="TODO: fatal error: execinfo.h: No such file or directory";;
+	#aarch64-musl) broken="TODO: fatal error: execinfo.h: No such file or directory";;
```
Comment (Piraty) [inline srcpkgs/faust/template]: please remove the comments, compilation works for those architectures (i compiled here locally, but didn't test the result)

Comment (DavideMaggio) [reply srcpkgs/faust/template]: ... sorry ... I'm doing mistakes with rebasing ... :( 
Can I close this pull request and reopen a new one?

Comment (DavideMaggio) [reply srcpkgs/faust/template]: ok, I removed the comments and squashed commits ... It's correct?

**Comment by ndowens** [comment]
Definitely wrong, including many people’s commits

**Comment by DavideMaggio** [comment]
sorry... I will reopen a clean pull request soon.

**Comment by Piraty** [comment]
Please don't close + reopen PRs just to "clean" them, but instead: clean your branch (squash your commits to conform the requirements as per manual, rebase to master) and force-push to your fork to update the existing PR.

rough guidance can be found in the [manual](https://github.com/void-linux/void-packages/blob/bb6228eb65f93686bcd3d6abbabcd60bf5bcbad1/Manual.md#contributing)

**Comment by DavideMaggio** [comment]
I know, I RTFM and I'm sorry ... :( 
I've already opened a new 'clean' PR: #17568, now we have two... can we close this one please?

**Comment by ndowens** [comment]
You are able to close your own PRs

**Pull request closed without merging**